### PR TITLE
chore: Update Switch to use griffel reset styles

### PR DESCRIPTION
**Pull request by behowell**
## Previous Behavior

Switch styles add a large number of classes at a default state. This is similar to Checkbox and Radio; see #25984 for a more detailed description.

## New Behavior

Update the styles to apply base styles with Griffe's makeResetStyles function, which combines all of the base styles into a single class.

## Related Issue(s)

This is a follow-up to this PR, which did the same thing for Checkbox and Radio:
* #25984

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                          |     Baseline (minified/GZIP) |                          PR |                                                                                                                                                                                                           Change |
| :----------------------------------------------------------------------------------------- | ---------------------------: | --------------------------: | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr> | `33.427 kB`<br />`10.581 kB` | `27.372 kB`<br />`8.453 kB` | `-6.055 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/decrease.png" /><br />`-2.128 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/decrease.png" /> |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `59.288 kB`<br />`16.435 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `186.264 kB`<br />`52.303 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |   `33.75 kB`<br />`11.101 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/983c8a8477ffb675d5c756b5407d926bb90a5ad7'>983c8a8477ffb675d5c756b5407d926bb90a5ad7</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/Avatar-mount_base.html">1612</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/Avatar-mount.html">1580</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/Button-mount_base.html">1106</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/Button-mount.html">1108</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/FluentProvider-mount_base.html">1905</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/FluentProvider-mount.html">1888</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/FluentProviderWithTheme-mount_base.html">775</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/FluentProviderWithTheme-mount.html">772</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">687</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">698</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">748</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">757</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/MakeStyles-mount_base.html">2309</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/MakeStyles-mount.html">2248</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/Persona-mount_base.html">3616</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/Persona-mount.html">3586</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/SpinButton-mount_base.html">3005</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26007/perf-test-react-components/SpinButton-mount.html">3026</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-switch/src/components/Switch/useSwitchStyles.ts [hunk 1]
```diff
@@ -22,175 +22,173 @@ const trackHeight = 20;
...
-    boxSizing: 'border-box',
-    fill: 'currentColor',
-    flexShrink: 0,
-    fontSize: `${thumbSize}px`,
-    height: `${trackHeight}px`,
-    ...shorthands.margin(tokens.spacingVerticalS, tokens.spacingHorizontalS),
-    pointerEvents: 'none',
+const useIndicatorBaseClassName = makeResetStyles({
+  borderRadius: tokens.borderRadiusCircular,
+  borderStyle: 'solid',
```
Comment (spmonahan) [inline packages/react-components/react-switch/src/components/Switch/useSwitchStyles.ts:42]: nit: `borderStyle` and `borderWidth` can be combined into a single rule.

File: packages/react-components/react-switch/src/components/Switch/useSwitchStyles.ts [hunk 2]
```diff
@@ -22,175 +22,173 @@ const trackHeight = 20;
...
+  borderWidth: '1px',
+  lineHeight: 0,
+  boxSizing: 'border-box',
+  fill: 'currentColor',
+  flexShrink: 0,
+  fontSize: `${thumbSize}px`,
+  height: `${trackHeight}px`,
+  margin: tokens.spacingVerticalS + ' ' + tokens.spacingHorizontalS,
+  pointerEvents: 'none',
+  transitionDuration: tokens.durationNormal,
```
Comment (spmonahan) [inline packages/react-components/react-switch/src/components/Switch/useSwitchStyles.ts:52]: nit: `transition*` properties can all be combined.

File: packages/react-components/react-switch/src/components/Switch/useSwitchStyles.ts [hunk 3]
```diff
@@ -22,175 +22,173 @@ const trackHeight = 20;
...
+  transitionTimingFunction: tokens.curveEasyEase,
+  transitionProperty: 'background, border, color',
+  width: `${trackWidth}px`,
+
+  '@media screen and (prefers-reduced-motion: reduce)': {
+    transitionDuration: '0.01ms',
```
Comment (spmonahan) [inline packages/react-components/react-switch/src/components/Switch/useSwitchStyles.ts:58]: This value seems like a potential candidate for a token.

**Review by spmonahan — APPROVED** [review]

**Review by khmakoto — APPROVED** [review]

File: packages/react-components/react-switch/src/components/Switch/useSwitchStyles.ts (hunk 3 above)
Comment (behowell) [reply packages/react-components/react-switch/src/components/Switch/useSwitchStyles.ts:58]: Probably, but there's no token for it at the moment as far as I can tell.

**Pull request merged**